server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/GetPartitioningAdditionalData.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.server.api.database.runs.functions
+
+import doobie.Fragment
+import doobie.implicits.toSqlInterpolator
+import doobie.util.Read
+import play.api.libs.json.Json
+import za.co.absa.atum.model.dto.PartitioningSubmitDTO
+import za.co.absa.atum.server.api.database.PostgresDatabaseProvider
+import za.co.absa.atum.server.api.database.runs.Runs
+import za.co.absa.atum.server.model.PartitioningForDB
+import za.co.absa.fadb.DBSchema
+import za.co.absa.fadb.doobie.DoobieFunction.DoobieMultipleResultFunction
+import za.co.absa.fadb.doobie.DoobieEngine
+import zio.interop.catz.asyncInstance
+import zio.{Task, URLayer, ZIO, ZLayer}
+
+import za.co.absa.atum.server.api.database.DoobieImplicits.getMapWithOptionStringValues
+
+class GetPartitioningAdditionalData (implicit schema: DBSchema, dbEngine: DoobieEngine[Task])
+  extends DoobieMultipleResultFunction[PartitioningSubmitDTO, (String, Option[String]), Task]
+  {
+
+    override def sql(values: PartitioningSubmitDTO)(implicit read: Read[(String, Option[String])]): Fragment = {
```

Comment:
`PartitioningSubmitDTO` is too broad and most things are not needed from this DTO - you are only using `partioning`, not `authorIfNew` or `parentPartitioning`. So please use something else, possibly `PartitioningDTO` from the model